libbeat/processors/add_process_metadata/add_process_metadata.go

```diff
@@ -177,9 +215,24 @@ func (p *addProcessMetadata) enrich(event common.MapStr, pidField string) (resul
 			return nil, err
 		}
 	}
+
 	return result, nil
 }
 
+// add container.id into meta for mapping to pickup
+func (p *addProcessMetadata) enrichContainerID(pid int, meta common.MapStr) (common.MapStr, error) {
```

Comment:
I don't think this needs to pass back the `meta` reference. Just have it return the `error`.